### PR TITLE
ENCD-3971 Update node npm to version 10

### PR DESCRIPTION
**Pull request by forresttanaka**
Update node/npm to the latest LTS versions (currently node 10.15.0 and npm 6.4.1).

Everyone who gets this branch’s changes _must_ update their node installation to these (or similar major) versions to do a build. How to upgrade depends on how you have installed node. I had a very non-standard installation that took a fair amount of work to update. I’m hoping people with manually or brew-installed npm will have a more straightforward upgrade.

I also included documentation for maintaining package.json and package-lock.json. package-lock.json is a newer name to the old npm-shrinkwrap.json, but it seems much easier to maintain.

### CODE REVIEW

**Comment by forresttanaka** [comment]
As I mentioned in one poorly-spelled commit, I included the change from the ENCD-XXXX-Travis-update-karen-winnt branch to make sure this change can pass Travis. I can back this out if preferred.

File: package-lock.json [hunk 1]
```diff
@@ -0,0 +1,11239 @@
+{
```
Comment (caseylitton) [inline package-lock.json]: New file or should we add this to git ignore?

File: .travis.yml [hunk 2]
```diff
@@ -80,7 +80,8 @@ script:
         --browser-arg-int build  "$TRAVIS_BUILD_NUMBER" \
         --browser-arg-int idleTimeout 300 \
         --browser-arg name "$TRAVIS_REPO_SLUG $TRAVIS_BRANCH $TRAVIS_COMMIT" \
-        --browser-arg browser "$BROWSER"
+        --browser-arg browser "$BROWSER" \
+        --browser-arg platform "Linux"
```
Comment (caseylitton) [inline .travis.yml]: As you mentioned...I think it's fine in this PR.  Maybe we'll want to separate via commits before merging.

File: .travis.yml [hunk 3]
```diff
@@ -80,7 +80,8 @@ script:
         --browser-arg-int build  "$TRAVIS_BUILD_NUMBER" \
         --browser-arg-int idleTimeout 300 \
         --browser-arg name "$TRAVIS_REPO_SLUG $TRAVIS_BRANCH $TRAVIS_COMMIT" \
-        --browser-arg browser "$BROWSER"
+        --browser-arg browser "$BROWSER" \
```
Comment (caseylitton) [inline .travis.yml]: remove since this is in origin/dev now.

**Pull request merged**